crates/net/downloaders/src/optimistic/buffer.rs

```diff
@@ -0,0 +1,70 @@
+use lru::LruCache;
+use reth_eth_wire::BlockBody;
+use reth_primitives::{SealedHeader, H256};
+use std::num::NonZeroUsize;
+
+/// The downloader buffer.
+#[derive(Debug)]
+pub(crate) struct DownloadBuffer {
+    // TODO: merge two buffers?
+    /// Internal header buffer
+    headers: LruCache<H256, SealedHeader>,
```

Comment:
What's the benefit of `LruCache` vs a `Vec` with a fixed capacity?

Reply:
maybe a `LruCache` vs a `HashMap` with a fixed capacity would be a better comparison. it's LRU for now, but I'm still thinking about whether it introduces additional complexity. the benefit is that the headers buffer is used to lookup downloaded headers for both headers and bodies requests